lib/matplotlib/collections.py

```diff
@@ -679,6 +679,14 @@ def set_alpha(self, alpha):
         """
         if alpha is not None:
             try:
+                # Check if alpha is an array of matching size
+                np_alpha = ma.asarray(alpha)
+                if np_alpha.ndim == 1:
+                    artist.Artist.set_alpha(self, np_alpha)
+                    return
+            except AttributeError:
```

Comment:
The try-except is likely unnecessary now.  I missed this, but I will remove it now.